src/qibolab/instruments/bluefors.py

```diff
@@ -0,0 +1,49 @@
+# import re
+import socket
+
+from qibo.config import log
+
+# from datetime import datetime
+
+
+class TemperatureController:
+    """Bluefors temperature controller."""
+
+    def __init__(self, ip_address: str, port: int = 8888):
+        """Creation of the controller object.
+
+        Args:
+            ip_address (str): IP address of the board sending cryo temperature data.
+            port (int): port of the board sending cryo temperature data.
+        """
+        self.ip_address = ip_address
+        self.port = port
+
+    def get_data(self) -> str:
+        """Connect to the socket and get temperature data.
+
+        Returns:
+            message (str): socket message in this format:
+                flange_name: {'temperature': <value(float)>, 'timestamp':<value(float)>}
+        """
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((self.ip_address, self.port))
+        log.info("Bluefors Temperature Controller Connected")
+
+        # This message is a string that looks like a json
+        # Shoud we convert it into a dictionary?
+        message = client_socket.recv(1024).decode()
+        return message
+
+    def read_data(self):
+        """Continously read data from the temperature controller."""
+        while True:
+            yield self.get_data()
```

Comment:
```suggestion
        while True:
            yield client_socket.recv(1024).decode()
```

Isn't this enough? Is there any reason to use a new connection every time?

Reply:
Yes, it is enough, but have two reasons:
1. It might be interesting to be able to get a single temperature value.
2. What would happen if for some reason the connection cannot be established at the beginning?

Additionally, I see no performance difference between the two approaches.

Reply:
> 2. What would happen if for some reason the connection cannot be established at the beginning?

In this case, you should get an error in both cases.

However, thinking again about it, I'd propose to have a `.connect()` method, like we have for the other instruments.

Then you can keep the current `.get_data()` and `.read_data()` as they are (just without the connection part in the first one).

Reply:
> However, thinking again about it, I'd propose to have a `.connect()` method, like we have for the other instruments.

Yes, this is also what I had in mind.

Reply:
> However, thinking again about it, I'd propose to have a `.connect()` method, like we have for the other instruments.

Yes, this is also what I had in mind.